Remove support for <attachment> thumbnails

hortont424 opened the pull request:
#### 0b57fc4f6093cac1bf3ef986098cd9570620b023
<pre>
Remove support for &lt;attachment&gt; thumbnails
<a href="https://bugs.webkit.org/show_bug.cgi?id=290976">https://bugs.webkit.org/show_bug.cgi?id=290976</a>

Reviewed by Wenson Hsieh and Gerald Squelart.

This code was disabled in 252627@main; if it ever comes back, it would have a very different shape.

* LayoutTests/TestExpectations:
* LayoutTests/fast/attachment/attachment-thumbnail-preserves-aspect-ratio-expected.txt: Removed.
* LayoutTests/fast/attachment/attachment-thumbnail-preserves-aspect-ratio.html: Removed.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::HTMLAttachmentElement::updateImage):
(WebCore::HTMLAttachmentElement::updateThumbnailForNarrowLayout): Deleted.
(WebCore::HTMLAttachmentElement::updateThumbnailForWideLayout): Deleted.
* Source/WebCore/html/HTMLAttachmentElement.h:
* Source/WebCore/rendering/AttachmentLayout.mm:
(WebCore::AttachmentLayout::AttachmentLayout):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::paintAttachmentIcon):
(WebCore::shouldDrawIcon): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::attachmentThumbnailInfo): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _insertAttachmentWithFileWrapper:contentType:completion:]):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::requestThumbnail): Deleted.
(WebKit::WebPageProxy::requestThumbnailWithPath): Deleted.
* Source/WebKit/UIProcess/QuickLookThumbnailLoader.h: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailLoader.mm: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailingSoftLink.h: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailingSoftLink.mm: Removed.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateAttachmentAttributes):
(WebKit::WebPageProxy::registerAttachmentIdentifierFromFilePath):
(WebKit::WebPageProxy::registerAttachmentsFromSerializedData):
(WebKit::WebPageProxy::updateAttachmentThumbnail): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateAttachmentThumbnail): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
(TestWebKitAPI::TEST(WKAttachmentTests, CutAndPasteAttachmentBetweenWebViews)):
(TestWebKitAPI::_generateBestRepresentationForRequest): Deleted.
(TestWebKitAPI::TEST(WKAttachmentTests, iWorkAttachmentWithoutPasteboardActionLoadsThumbnail)): Deleted.

Canonical link: <a href="https://commits.webkit.org/293181@main">https://commits.webkit.org/293181@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6ba8f4ed915b1969aa68e78e493ef33e090cdca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98145 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8003 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103261 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/48674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100190 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26227 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/74725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/48674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101149 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13692 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/88680 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55085 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/13476 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6614 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/48116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/83453 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6694 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105638 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25231 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18383 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/83711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25604 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84861 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83164 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21001 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27804 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/5489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/18871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25190 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/30364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25010 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28326 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26585 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

squelart commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1436,38 +1436,15 @@ static void paintAttachmentIconBackground(const RenderAttachment& attachment, Gr
 
-static bool shouldDrawIcon(const String& title)
-{
-#if HAVE(QUICKLOOK_THUMBNAILING)
```
I see you're removing all the `QUICKLOOK_THUMBNAILING` uses.
I think you missed the ones in Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm .
You may want to do a find-all, in case there are more stragglers.

hortont424 replied:
Good call!

squelart reviewed (approved):
Nice code deletion, thank you!
Just one missing file to clean up, I think.

hortont424 committed:
Remove support for <attachment> thumbnails
https://bugs.webkit.org/show_bug.cgi?id=290976

Reviewed by Wenson Hsieh and Gerald Squelart.

This code was disabled in 252627@main; if it ever comes back, it would have a very different shape.

* LayoutTests/TestExpectations:
* LayoutTests/fast/attachment/attachment-thumbnail-preserves-aspect-ratio-expected.txt: Removed.
* LayoutTests/fast/attachment/attachment-thumbnail-preserves-aspect-ratio.html: Removed.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::HTMLAttachmentElement::updateImage):
(WebCore::HTMLAttachmentElement::updateThumbnailForNarrowLayout): Deleted.
(WebCore::HTMLAttachmentElement::updateThumbnailForWideLayout): Deleted.
* Source/WebCore/html/HTMLAttachmentElement.h:
* Source/WebCore/rendering/AttachmentLayout.mm:
(WebCore::AttachmentLayout::AttachmentLayout):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::paintAttachmentIcon):
(WebCore::shouldDrawIcon): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::attachmentThumbnailInfo): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _insertAttachmentWithFileWrapper:contentType:completion:]):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::requestThumbnail): Deleted.
(WebKit::WebPageProxy::requestThumbnailWithPath): Deleted.
* Source/WebKit/UIProcess/QuickLookThumbnailLoader.h: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailLoader.mm: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailingSoftLink.h: Removed.
* Source/WebKit/UIProcess/QuickLookThumbnailingSoftLink.mm: Removed.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateAttachmentAttributes):
(WebKit::WebPageProxy::registerAttachmentIdentifierFromFilePath):
(WebKit::WebPageProxy::registerAttachmentsFromSerializedData):
(WebKit::WebPageProxy::updateAttachmentThumbnail): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateAttachmentThumbnail): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
(TestWebKitAPI::TEST(WKAttachmentTests, CutAndPasteAttachmentBetweenWebViews)):
(TestWebKitAPI::_generateBestRepresentationForRequest): Deleted.
(TestWebKitAPI::TEST(WKAttachmentTests, iWorkAttachmentWithoutPasteboardActionLoadsThumbnail)): Deleted.

Canonical link: https://commits.webkit.org/293181@main
